ocs_ci/helpers/helpers.py

```diff
@@ -4486,3 +4493,108 @@ def verify_log_exist_in_pods_logs(
         if expected_log in pod_logs:
             return True
     return False
+
+
+def retrieve_cli_binary(cli_type="mcg"):
+    """
+    Download the MCG-CLI/ODF-CLI binary and store it locally.
+
+    Args:
+        cli_type (str): choose which bin file you want to download ["odf" -> odf-cli , "mcg" -> mcg-cli]
+
+    Raises:
+        AssertionError: In the case the CLI binary is not executable.
+
+    """
+    semantic_version = version.get_semantic_ocs_version_from_config()
+    if cli_type == "odf" and semantic_version < version.VERSION_4_15:
+        raise NotSupportedException(
+            f"odf cli tool not supported on ODF {semantic_version}"
+        )
+
+    remote_path = get_architecture_path(cli_type)
+    remote_cli_basename = os.path.basename(remote_path)
+    if cli_type == "mcg":
+        local_cli_path = constants.NOOBAA_OPERATOR_LOCAL_CLI_PATH
+    elif cli_type == "odf":
+        local_cli_path = constants.CLI_TOOL_LOCAL_PATH
+    local_cli_dir = os.path.dirname(local_cli_path)
+    live_deployment = config.DEPLOYMENT["live_deployment"]
+    if semantic_version >= version.VERSION_4_15:
+        if live_deployment:
+            image = f"{constants.ODF_CLI_OFFICIAL_IMAGE}:v{semantic_version}"
+        else:
+            image = f"{constants.ODF_CLI_DEV_IMAGE}:v{semantic_version}"
+    elif semantic_version >= version.VERSION_4_13:
+        if live_deployment:
+            image = f"{constants.MCG_CLI_OFFICIAL_IMAGE}:v{semantic_version}"
+        else:
+            image = f"{constants.MCG_CLI_DEV_IMAGE}:v{semantic_version}"
+    else:
+        image = f"{constants.MCG_CLI_DEV_IMAGE}:v{semantic_version}"
```

Comment:
It looks like you're choosing between the ODF_CLI image and MCG_CLI image here based on the version of the cluster instead of the `cli_type` argument. The result is the following error when attempting to run a test that needs to download the mcg-cli:

```
ocs_ci.ocs.exceptions.CommandFailed: Error during execution of command: oc image extract --registry-config /my/path/to/ocs-ci/data/pull-secret quay.io/rhceph-dev/odf4-odf-cli-rhel9:v4.15 --confirm --path /my/path/to/ocs-ci/data.

Error is error: image "quay.io/rhceph-dev/odf4-odf-cli-rhel9:v4.15" not found: manifest unknown: manifest unknown
```

Reply:
@sagihirshfeld  It is failed because the image `quay.io/rhceph-dev/odf4-odf-cli-rhel9:v4.15` does not exist.
I fixed it by using mcg cli registry.